ui/constants/emotes.js

```diff
@@ -90,7 +90,929 @@ const getEmotes = (px: string, multiplier: string) => [
+
+        buildTwemote('flag_uzbekistan', `flags/flag_uzbekistan.png`, '🇺🇿'),
+        buildTwemote('flag_vatican_city', `flags/flag_vatican_city.png`, '🇻🇦'),
+        buildTwemote('flag_st_vincent_grenadines', `flags/flag_st_vincent_grenadines.png`, '🇻🇨'),
+        buildTwemote('flag_venezuela', `flags/flag_venezuela.png`, '🇻🇪'),
+        buildTwemote('flag_british_virgin_islands', `flags/flag_british_virgin_islands.png`, '🇻🇬'),
+        buildTwemote('flag_us_virgin_islands', `flags/flag_us_virgin_islands.png`, '🇻🇮'),
+        buildTwemote('flag_vietnam', `flags/flag_vietnam.png`, '🇻🇳'),
+
+        buildTwemote('flag_vanuatu', `flags/flag_vanuatu.png`, '🇻🇺'),
+        buildTwemote('flag_wallis_futuna', `flags/flag_wallis_futuna.png`, '🇼🇫'),
+        buildTwemote('flag_samoa', `flags/flag_samoa.png`, '🇼🇸'),
+        buildTwemote('flag_kosovo', `flags/flag_kosovo.png`, '🇽🇰'),
+        buildTwemote('flag_yemen', `flags/flag_yemen.png`, '🇾🇪'),
+        buildTwemote('flag_mayotte', `flags/flag_mayotte.png`, '🇾🇹'),
+        buildTwemote('flag_south_africa', `flags/flag_south_africa.png`, '🇿🇦'),
+
+        buildTwemote('flag_zambia', `flags/flag_zambia.png`, '🇿🇲'),
+        buildTwemote('flag_zimbabwe', `flags/flag_zimbabwe.png`, '🇿🇼'),
+        buildTwemote('flag_england', `flags/flag_england.png`, '🏴󠁧󠁢󠁥󠁮󠁧󠁿'),
+        buildTwemote('flag_scotland', `flags/flag_scotland.png`, '🏴󠁧󠁢󠁳󠁣󠁴󠁿'),
+        buildTwemote('flag_wales', `flags/flag_wales.png`, '🏴󠁧󠁢󠁷󠁬󠁳󠁿'),
+
+        buildTwemote('flag_european_union', `flags/flag_european_union.png`, '🇪🇺'),
+        buildTwemote('flag_united_nations', `flags/flag_united_nations.png`, '🇺🇳'),
+
+        buildTwemote('pirate_flag', `flags/pirate_flag.png`, '🏴‍☠️'),
+        buildTwemote('rainbow_flag', `flags/rainbow_flag.png`, '🏳️‍🌈'),
+        buildTwemote('transgender_flag', `flags/transgender_flag.png`, '🏳️‍⚧️'),
+        buildTwemote('kek_flag', `flags/kek_flag.png`, ':kek_flag:'),
+        buildTwemote('anarchocapitalism_flag', `flags/anarchocapitalism_flag.png`, ':anarchocapitalism_flag:'),
+        buildTwemote('communism_flag', `flags/communism_flag.png`, ':communism_flag:'),
+        buildTwemote('antifa_flag', `flags/antifa_flag.png`, ':antifa_flag:'),
+        buildTwemote('gadsden_flag', `flags/gadsden_flag.png`, ':gadsden_flag:'),
+        buildTwemote('white_flag', `flags/white_flag.png`, '🏳'),
+        buildTwemote('black_flag', `flags/black_flag.png`, '🏴'),
+      ];
+  }
+};
+
+function buildTwemoteArray(): Array<Object> {
+  return []
+    .concat(getTwemotes('SMILIES'))
+    .concat(getTwemotes('HANDSIGNALS'))
+    .concat(getTwemotes('ACTIVITIES'))
+    .concat(getTwemotes('SYMBOLS'))
+    .concat(getTwemotes('NATURE'))
+    .concat(getTwemotes('FOOD'))
+    .concat(getTwemotes('FLAGS'));
+}
+
+function buildTwemoteNameArray() {
+  let res = [];
+  for (let entry of buildTwemoteArray()) {
+    // $FlowIgnore
+    res.push(entry.name);
+  }
+  return res;
+}
```

Comment:
:small_blue_diamond: Instead of calling `buildTweoteArray` again for the `for` loop, this function can take in a parameter (passing in the generated `TWEMOTEARRAY`) so we can avoid building twice (I think 3 times, based on what I'm seeing)

Reply:
But this is minor ... the overall time is ~1s on debug build

Reply:
Good point. I also did something similar with `buildTwemoteArray`.